In [77]:
import pandas as pd
import numpy as np
import matplotlib.pyplot as plt
import seaborn as sns

# Display plots inside notebook
%matplotlib inline

In [78]:
print("Pandas version:", pd.__version__)

Pandas version: 3.0.3


In [79]:
import pandas as pd

daily_activity = pd.read_csv("../data/raw/dailyActivity_merged.csv")
sleep_day = pd.read_csv("../data/raw/sleepDay_merged.csv")
heart_rate = pd.read_csv("../data/raw/heartrate_seconds_merged.csv")
weight = pd.read_csv("../data/raw/weightLogInfo_merged.csv")
sleep_lifestyle = pd.read_csv("../data/raw/Sleep_health_and_lifestyle_dataset.csv")

FileNotFoundError: [Errno 2] No such file or directory: '../data/raw/dailyActivity_merged.csv'

In [ ]:
import os

print("Current folder:", os.getcwd())

In [ ]:
2+2


4

In [ ]:
import pandas as pd

daily_activity = pd.read_csv("data/raw/dailyActivity_merged.csv")
sleep_day = pd.read_csv("data/raw/sleepDay_merged.csv")
heart_rate = pd.read_csv("data/raw/heartrate_seconds_merged.csv")
weight = pd.read_csv("data/raw/weightLogInfo_merged.csv")
sleep_lifestyle = pd.read_csv("data/raw/Sleep_health_and_lifestyle_dataset.csv")

print("All datasets loaded successfully")

All datasets loaded successfully


In [ ]:
print("Daily Activity:", daily_activity.shape)
print("Sleep Day:", sleep_day.shape)
print("Heart Rate:", heart_rate.shape)
print("Weight:", weight.shape)
print("Sleep Lifestyle:", sleep_lifestyle.shape)

Daily Activity: (457, 15)
Sleep Day: (413, 5)
Heart Rate: (1154681, 3)
Weight: (67, 8)
Sleep Lifestyle: (374, 13)


In [ ]:
daily_activity.head()

,Id,ActivityDate,TotalSteps,TotalDistance,TrackerDistance,LoggedActivitiesDistance,VeryActiveDistance,ModeratelyActiveDistance,LightActiveDistance,SedentaryActiveDistance,VeryActiveMinutes,FairlyActiveMinutes,LightlyActiveMinutes,SedentaryMinutes,Calories
0,1503960366,3/25/2016,11004,7.11,7.11,0.0,2.57,0.46,4.07,0.0,33,12,205,804,1819
1,1503960366,3/26/2016,17609,11.55,11.55,0.0,6.92,0.73,3.91,0.0,89,17,274,588,2154
2,1503960366,3/27/2016,12736,8.53,8.53,0.0,4.66,0.16,3.71,0.0,56,5,268,605,1944
3,1503960366,3/28/2016,13231,8.93,8.93,0.0,3.19,0.79,4.95,0.0,39,20,224,1080,1932
4,1503960366,3/29/2016,12041,7.85,7.85,0.0,2.16,1.09,4.61,0.0,28,28,243,763,1886


In [80]:
daily_activity.info()
sleep_day.info()
heart_rate.info()
weight.info()
sleep_lifestyle.info()

<class 'pandas.DataFrame'>
RangeIndex: 457 entries, 0 to 456
Data columns (total 15 columns):
 #   Column                    Non-Null Count  Dtype  
---  ------                    --------------  -----  
 0   Id                        457 non-null    int64  
 1   ActivityDate              457 non-null    str    
 2   TotalSteps                457 non-null    int64  
 3   TotalDistance             457 non-null    float64
 4   TrackerDistance           457 non-null    float64
 5   LoggedActivitiesDistance  457 non-null    float64
 6   VeryActiveDistance        457 non-null    float64
 7   ModeratelyActiveDistance  457 non-null    float64
 8   LightActiveDistance       457 non-null    float64
 9   SedentaryActiveDistance   457 non-null    float64
 10  VeryActiveMinutes         457 non-null    int64  
 11  FairlyActiveMinutes       457 non-null    int64  
 12  LightlyActiveMinutes      457 non-null    int64  
 13  SedentaryMinutes          457 non-null    int64  
 14  Calories             

In [81]:
daily_activity.isnull().sum()

Id                          0
ActivityDate                0
TotalSteps                  0
TotalDistance               0
TrackerDistance             0
LoggedActivitiesDistance    0
VeryActiveDistance          0
ModeratelyActiveDistance    0
LightActiveDistance         0
SedentaryActiveDistance     0
VeryActiveMinutes           0
FairlyActiveMinutes         0
LightlyActiveMinutes        0
SedentaryMinutes            0
Calories                    0
dtype: int64

In [82]:
sleep_day.isnull().sum()

Id                    0
SleepDay              0
TotalSleepRecords     0
TotalMinutesAsleep    0
TotalTimeInBed        0
dtype: int64

In [83]:
weight.isnull().sum()

Id                 0
Date               0
WeightKg           0
WeightPounds       0
Fat               65
BMI                0
IsManualReport     0
LogId              0
dtype: int64

In [84]:
sleep_lifestyle.isnull().sum()

Person ID                    0
Gender                       0
Age                          0
Occupation                   0
Sleep Duration               0
Quality of Sleep             0
Physical Activity Level      0
Stress Level                 0
BMI Category                 0
Blood Pressure               0
Heart Rate                   0
Daily Steps                  0
Sleep Disorder             219
dtype: int64

In [85]:
heart_rate.isnull().sum()

Id       0
Time     0
Value    0
dtype: int64

Duplicate records

In [86]:
print("Daily Activity:", daily_activity.duplicated().sum())
print("Sleep Day:", sleep_day.duplicated().sum())
print("Heart Rate:", heart_rate.duplicated().sum())
print("Weight:", weight.duplicated().sum())
print("Sleep Lifestyle:", sleep_lifestyle.duplicated().sum())

Daily Activity: 0
Sleep Day: 3
Heart Rate: 0
Weight: 0
Sleep Lifestyle: 0


SUMMARY STATISTICS

In [87]:
daily_activity.describe()

,Id,TotalSteps,TotalDistance,TrackerDistance,LoggedActivitiesDistance,VeryActiveDistance,ModeratelyActiveDistance,LightActiveDistance,SedentaryActiveDistance,VeryActiveMinutes,FairlyActiveMinutes,LightlyActiveMinutes,SedentaryMinutes,Calories
count,4.570000e+02,457.000000,457.000000,457.000000,457.000000,457.000000,457.000000,457.000000,457.000000,457.000000,457.000000,457.000000,457.000000,457.000000
mean,4.628595e+09,6546.562363,4.663523,4.609847,0.179427,1.180897,0.478643,2.890197,0.001904,16.623632,13.070022,170.070022,995.282276,2189.452954
std,2.293781e+09,5398.493064,4.082072,4.068540,0.849232,2.487159,0.830995,2.237523,0.008487,28.919704,36.208635,122.205372,337.021404,815.484523
min,1.503960e+09,0.000000,0.000000,0.000000,0.000000,0.000000,0.000000,0.000000,0.000000,0.000000,0.000000,0.000000,32.000000,0.000000
25%,2.347168e+09,1988.000000,1.410000,1.280000,0.000000,0.000000,0.000000,0.870000,0.000000,0.000000,0.000000,64.000000,728.000000,1776.000000
50%,4.057193e+09,5986.000000,4.090000,4.090000,0.000000,0.000000,0.020000,2.930000,0.000000,0.000000,1.000000,181.000000,1057.000000,2062.000000
75%,6.391747e+09,10198.000000,7.160000,7.110000,0.000000,1.310000,0.670000,4.460000,0.000000,25.000000,16.000000,257.000000,1285.000000,2667.000000
max,8.877689e+09,28497.000000,27.530001,27.530001,6.727057,21.920000,6.400000,12.510000,0.100000,202.000000,660.000000,720.000000,1440.000000,4562.000000


### Observation: Daily Activity Dataset

The Daily Activity dataset contains 457 records. The average daily step count is approximately 6,547 steps, while the median is 5,986 steps. The average calories burned is around 2,189 calories per day. Some records show 0 steps and 0 calories, which may need further investigation during preprocessing. Sedentary minutes are high, with an average of about 995 minutes per day.

In [88]:
sleep_day.describe()

,Id,TotalSleepRecords,TotalMinutesAsleep,TotalTimeInBed
count,4.130000e+02,413.000000,413.000000,413.000000
mean,5.000979e+09,1.118644,419.467312,458.639225
std,2.060360e+09,0.345521,118.344679,127.101607
min,1.503960e+09,1.000000,58.000000,61.000000
25%,3.977334e+09,1.000000,361.000000,403.000000
50%,4.702922e+09,1.000000,433.000000,463.000000
75%,6.962181e+09,1.000000,490.000000,526.000000
max,8.792010e+09,3.000000,796.000000,961.000000


### Observation: Sleep Day Dataset

The Sleep Day dataset contains 413 sleep records. Participants slept for an average of approximately 419 minutes (about 7 hours) per night, while the average time spent in bed was around 459 minutes (about 7.6 hours). This indicates that participants spent approximately 40 minutes awake while in bed on average. Most participants had one sleep record per day, with a maximum of three sleep records recorded for a single day.

In [89]:
heart_rate.describe()

,Id,Value
count,1.154681e+06,1.154681e+06
mean,5.352122e+09,7.975687e+01
std,2.033584e+09,1.872988e+01
min,2.022484e+09,3.600000e+01
25%,4.020333e+09,6.600000e+01
50%,5.553957e+09,7.700000e+01
75%,6.962181e+09,9.000000e+01
max,8.877689e+09,1.850000e+02


### Observation: Heart Rate Dataset

The Heart Rate dataset contains 1,154,681 heart rate measurements collected at one-second intervals. The average heart rate is approximately 80 beats per minute (bpm), with a minimum value of 36 bpm and a maximum value of 185 bpm. The large number of observations makes this the largest dataset in the project and provides detailed physiological information for analysing user activity and health patterns.

In [90]:
weight.describe()

,Id,WeightKg,WeightPounds,Fat,BMI,LogId
count,6.700000e+01,67.000000,67.000000,2.00000,67.000000,6.700000e+01
mean,7.009282e+09,72.035821,158.811801,23.50000,25.185224,1.461772e+12
std,1.950322e+09,13.923206,30.695415,2.12132,3.066963,7.829948e+08
min,1.503960e+09,52.599998,115.963147,22.00000,21.450001,1.460444e+12
25%,6.962181e+09,61.400002,135.363832,22.75000,23.959999,1.461079e+12
50%,6.962181e+09,62.500000,137.788914,23.50000,24.389999,1.461802e+12
75%,8.877689e+09,85.049999,187.503152,24.25000,25.559999,1.462375e+12
max,8.877689e+09,133.500000,294.317120,25.00000,47.540001,1.463098e+12


### Observation: Weight Dataset

The Weight dataset contains 67 records of participants' body weight and BMI measurements. The average body weight is approximately 72.04 kg, with an average BMI of 25.19, which falls within the overweight category according to standard BMI classifications. The `Fat` column contains data for only two records, making it unsuitable for further analysis. Therefore, this column will be removed during the data preprocessing stage.

In [91]:
sleep_lifestyle.describe()

,Person ID,Age,Sleep Duration,Quality of Sleep,Physical Activity Level,Stress Level,Heart Rate,Daily Steps
count,374.000000,374.000000,374.000000,374.000000,374.000000,374.000000,374.000000,374.000000
mean,187.500000,42.184492,7.132086,7.312834,59.171123,5.385027,70.165775,6816.844920
std,108.108742,8.673133,0.795657,1.196956,20.830804,1.774526,4.135676,1617.915679
min,1.000000,27.000000,5.800000,4.000000,30.000000,3.000000,65.000000,3000.000000
25%,94.250000,35.250000,6.400000,6.000000,45.000000,4.000000,68.000000,5600.000000
50%,187.500000,43.000000,7.200000,7.000000,60.000000,5.000000,70.000000,7000.000000
75%,280.750000,50.000000,7.800000,8.000000,75.000000,7.000000,72.000000,8000.000000
max,374.000000,59.000000,8.500000,9.000000,90.000000,8.000000,86.000000,10000.000000


### Observation: Sleep Lifestyle Dataset

The Sleep Lifestyle dataset contains 374 participant records. The average age of participants is approximately 42 years. On average, participants sleep 7.13 hours per night and report a sleep quality score of 7.31 out of 10. The average daily physical activity level is 59.17, while the average stress level is 5.39. Participants have an average resting heart rate of 70 beats per minute and take approximately 6,817 steps per day. This dataset provides valuable lifestyle and health information that can be used to analyse factors affecting sleep quality and overall well-being.